app/src/main/java/io/github/droidkaigi/confsched2018/presentation/sessions/AllSessionsViewModel.kt

```diff
@@ -26,6 +26,9 @@ class AllSessionsViewModel @Inject constructor(
     private val focusCurrentSession: MutableLiveData<Boolean> = MutableLiveData()
     val refreshFocusCurrentSession: LiveData<Boolean> = focusCurrentSession
 
+    private val restorePreviousSession: MutableLiveData<Boolean> = MutableLiveData()
```

Comment:
I have a question 🙋 
Why this property need? Can we do it without this?

Reply:
Ahhh...
In conclusion, maybe I can do it.
I explain the process.

The sessionsViewModel is set twice.
The one by the Room DB, then another by the network API.
So if implemented as below, doesn't using the property, the previous position was restored twice. 
```
sessionsViewModel.sessions.observe(this, { result ->
    when (result) {
        is Result.Success -> {
            val sessions = result.data
            sessionsSection.updateSessions(sessions, onFavoriteClickListener,
                    onFeedbackListener, true)

            sessionsViewModel.onSuccessFetchSessions()
            **scrollToPreviousSession()**
        }
        is Result.Failure -> {
            Timber.e(result.e)
        }
    }
})
```
So I maked the property for not calling twice it. (I mimicked ** refreshFocusCurrentSession**)

However, currently, previousSessionPrefs will be initialized when restore succeded.
In addition,  if saved position is '-1', this feature has no operation.

Therefore, I think it is needless. 
Sorry, I will challenge to fix. 💪

Reply:
Excuse me, I want to discuss this.
I said "it is needless", but I reviewed my code, I think that it may be need.
All/Room/ScheduleFragments can't get own Tab ID, so they can't judge doing restoration or not.
Therefore SessionsFragment need to inform to them that it is target or not.
However the inform timing is out of sync with the getting sessions timing of All/Room/ScheduleFragments.

So I set an observer to the restoration target Fragment as below from the `SessionsFragment#reopenPreviousOpenedItem()` in current implemantation.

```
sessionsViewModel.reopenPreviousSession.observe(this, {
    if (it != true) return@observe
    scrollToPreviousSession()
})
```
I am not thinking also this is the best...but I don't have another idea.
Is there is a good idea?

Reply:
OK I understood 👍 
How about this? 🤔 
Please create a property in the ViewModel. 

```kotlin
val isNeedRestoreScrollState : Boolean = false
```

Please change restorePreviousSession to requestRestoringPreviousSession
in requestRestoringPreviousSession()

```kotlin
if (sessionsViewModel.sessions is Result.SUCCESS) {
  scrollToPreviousSession()
} else {
  sessionsViewModel.isNeedRestoreScrollState = true
}
```

Please change onSuccessFetchSessions to onShowSessions(Because it is view event.)

https://github.com/DroidKaigi/conference-app-2018/pull/605/files#diff-6d4ba3fe24616173b16fd16956439d62R89

```kotlin
if (sessionsViewModel.isNeedRestoreScrollState) {
  scrollToPreviousSession()
}
```

```
private fun scrollToPreviousSession() {
  sessionsViewModel.isNeedRestoreScrollState = false
...
```

Reply:
Thank you very much!!
I try it.

Reply:
Sorry please use requestRestoringScrollState instead of requestRestoringPreviousSession 🙇